tests/ops/qubit/test_hamiltonian.py

```diff
@@ -825,6 +825,12 @@ def test_compare_gell_mann(self):
         assert H2.compare(qml.GellMann(wires=2, index=2) @ qml.GellMann(wires=1, index=2)) is False
         assert H2.compare(H4) is False
 
+    def test_hamiltonian_compare():
+        """to test if hamiltonian comapre function has float point error"""
+        H2 = qml.Hamiltonian([0.3, 0.3], [qml.PauliZ(0), qml.PauliZ(0)])
+        H1 = qml.Hamiltonian([0.1, 0.2, 0.3], [qml.PauliZ(0), qml.PauliZ(0), qml.PauliZ(0)])
+        assert H1.compare(H2)
+
```

Comment:
The test looks good. However, some more tests should be added:
* Test that comparing non-equivalent Hamiltonians returns false.
* Test comparison between a Hamiltonian and a Tensor (both true and false cases).
* Test comparison between a Hamiltonian and "vanilla observable (eg. `qml.PauliX`, etc., both true and false cases)

Reply:
Thank you for your reply @mudit2812. I don't know how the import functions got added, probably messing around with pylint or something did it. I will try updating the PR as per your request. Thanks.

Reply:
Thanks!